src/main/java/uk/gov/hmcts/reform/idam/web/config/OIDCLocaleChangeInterceptor.java

```diff
@@ -0,0 +1,77 @@
+package uk.gov.hmcts.reform.idam.web.config;
+
+import com.google.common.collect.ImmutableSet;
+import lombok.NonNull;
+import org.springframework.web.servlet.LocaleResolver;
+import org.springframework.web.servlet.i18n.LocaleChangeInterceptor;
+import org.springframework.web.servlet.support.RequestContextUtils;
+
+import javax.annotation.Nullable;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.validation.constraints.NotNull;
+import java.util.Locale;
+import java.util.Set;
+
+public class OIDCLocaleChangeInterceptor extends LocaleChangeInterceptor {
+
+    protected final Set<Locale> allowedLocales;
+
+    public OIDCLocaleChangeInterceptor(@NonNull final Set<Locale> allowedLocales) {
+        super();
+        this.allowedLocales = ImmutableSet.copyOf(allowedLocales);
+    }
+
+    /**
+     * Checks the defined parameter according to OIDC rules for {@code ui_locales}. See https://openid.net/specs/openid-connect-core-1_0.html.
+     * {@inheritDoc}
+     *
+     * @should set locale to a single tag
+     * @should accept no parameter
+     * @should accept invalid locales
+     * @should set locale to first matching language tag
+     */
+    @Override
+    public boolean preHandle(final HttpServletRequest request, final HttpServletResponse response, Object handler) {
```

Comment:
This is a modified code from the parent class. The change is to tokenize the parameter and process each language separately.